packages/cli/src/util/link/repo.ts

```diff
@@ -288,14 +288,9 @@ export function findProjectsFromPath(
         normalizedPath.startsWith(`${project.directory}/`)
       );
     });
-}
-
-/**
- * TODO: remove
- */
-export function findProjectFromPath(
-  projects: RepoProjectConfig[],
-  path: string
-): RepoProjectConfig | undefined {
-  return findProjectsFromPath(projects, path)[0];
+  // If there are multiple matches, we only want the most relevant
+  // selections (with the deepest directory depth), so pick the first
+  // one and filter on those matches.
+  const firstMatch = matches[0];
```

Comment:
Does this guarantee that `matches.length > 0`?

Reply:
No, but it's ok because then the `filter()` below would not be executed in the case of an empty array.